aiida/daemon/workflowmanager.py

```diff
@@ -40,10 +40,14 @@ def execute_steps():
     logger.info("Querying the worflow DB")
     
     running_steps = get_all_running_steps()
-    
+
     for s in running_steps:
         w = s.parent.get_aiida_class()
```

Comment:
I know this might seem unrelated, but can we avoid at this stage to get the aiida_class (at least until when we set the state to finished below and continue; after is ok)?
A couple of past issues (including recent emails on the mailing list) where partially related to old steps in the DB being still 'running', but this would fail (because the workflow code was removed), blocking the execution of _all_ workflows.